README
NAME
    SeeAlso::Source::BeaconAggregator - Beacon files as source for
    SeeAlso::Server

    SeeAlso::Source::BeaconAggregator::Maintenance - Administrative
    tasks for the collection of beacon files

    sasbactrl - Command line interface to the above

    SeeAlso::Source::BeaconAggregator::Publisher - Extended
    methods (act as a resolver, publish a beacon file for this
    collection)


DESCRIPTION
    This Module allows a collection of BEACON files (cf.
    http://de.wikipedia.org/wiki/Wikipedia:BEACON) to be used as
    SeeAlso::Source (probably in the context of an SeeAlso::Server
    application). Therefore it implements the four methods documented in
    SeeAlso::Source

    The BEACON files (lists of non-local identifiers of a certain type
    documenting the coverage of a given online database plus means for
    access) are imported by the methods provided by
    SeeAlso::Source::BeaconAggregator::Maintenance.pm, usually by employing
    the script sasbactrl.pl as command line client.

    Serving other formats than SeeAlso or providing a BEACON file with
    respect to this SeeAlso service is achieved by using
    SeeAlso::Source::BeaconAggregator::Publisher.

TROUBLESHOOTING

    In order to operate with SeeAlso::Identifier::GND 054 you'll have
    to work around the inconvenience of that module not being completely
    compliant to the Interface set up by SeeAlso::Identifier.
    A Q&D fix would be to add the follogwing two lines in your code:

*SeeAlso::Identifier::GND::canonical = \&SeeAlso::Identifier::GND::normalized;
*SeeAlso::Identifier::GND::hash = \&SeeAlso::Identifier::GND::indexed;


AUTHOR
    Thomas Berger <ThB@gymel.com>

COPYRIGHT
    This program is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself.

    The full text of the license can be found in the LICENSE file included
    with this module.



Changes
Revision history for Perl module SeeAlso::Source::BeaconAggregator

0.02_92 2016-06-18
    - issues with perl 5.22, SQlite > 3.7.9
    - autodectection of #TARGET meaning in files like those generated by
      < http://tools.wmflabs.org/wikidata-todo/beacon.php >

0.02_86 2013-12-16
    - load filters

0.02_75 2012-11-13
    - more intelligently reuse global counts
    - repeatable --pragma args for sasbactrl (get / set: simply piped thru)
    - tests for sasbactrl (need optional Test::Command::Simple)

0.02_72 2012-09-28
    - cache global counts in admin table

0.02_70 2012-09-28
    - cluster property for new() method allows to specify a database
      with identifiers to be considered equivalent
    - this database shold be setup with "prepareRedirs" on init() 
      [admin table property REDIRECTION_INDEX]

0.02_64 2012-06-14
    - autoIdentifier method (creates object of class identifierClass,
      or initializes from admin table property IDENTIFIER_CLASS)
    - be careful: new() for the blessed-scalar identifier classes 
      usually does not return a true value

0.02_57 2011-09-14
    - aliasfilter
    - TODO: analyze conflicting identifier classes

0.02 2011-05-22
    - Bug fixes
    - eliminated most fetchrow_array in favor of fetchrow_arraryref
      because of strange failures under Cygwin (DBD::SQLite 1.29,
      DBI 1.609, Engine 3.6.22)
    - Tests

0.01 Sat Nov 20 14:14:50 2010
    - original version; created by ExtUtils::ModuleMaker 0.51




Build.PL
use Module::Build;
# See perldoc Module::Build for details of how this works

my $build = Module::Build->new(
	module_name => 'SeeAlso::Source::BeaconAggregator',
	requires => {
			  perl	=> '5.8.2',
			  Carp	=> 0,
		    File::Temp	=> 0,
			   CGI	=> '3.0',
		    HTTP::Date	=> '1.46',    # '5.0',
		 HTTP::Request	=> '1.40',    # '5.0',
			   LWP	=> '5.802',
		LWP::UserAgent	=> '2.031',
			   DBI	=> '1.57',
		   DBD::SQLite	=> '1.25',
	       SeeAlso::Server	=> '0.57',
	       SeeAlso::Source	=> '0.56',
	     SeeAlso::Response	=> '0.54',
	   SeeAlso::Identifier	=> '0.46',
		},
	build_requires => {
      SeeAlso::Identifier::GND	=> '0.54',
		    Test::More	=> '0.94',
		     URI::file	=> '1.0',
#         Test::Command::Simple	=> 0,
          Test::Command::Simple	=> 0,
		},
   configure_requires => {
		 Module::Build  => '0.35',
                },
	recommends => {
			   LWP	=> '5.827',
		  Getopt::Long	=> 0,
		    Pod::Usage	=> 0,
	       SeeAlso::Server	=> '0.58',
      SeeAlso::Identifier::GND	=> '0.55',
      SeeAlso::Identifier::PND	=> '0.57',
		},
#	script_files => [
#	'bin/sasbactrl',
#		],
	license => 'perl',
	create_makefile_pl => 'traditional',
	meta_merge => {
          resources => {
            repository => 'https://github.com/gymel/sasba.git',
            bugtracker => 'htps://github.com/gymel/sasba/issues',
            homepage => 'https://github.com/gymel/sasba',
          }
        }
    );

$build->create_build_script();


Makefile.PL
# Note: this file was auto-generated by Module::Build::Compat version 0.4218
require 5.008002;
use ExtUtils::MakeMaker;
WriteMakefile
(
  'PL_FILES' => {},
  'NAME' => 'SeeAlso::Source::BeaconAggregator',
  'VERSION_FROM' => 'lib/SeeAlso/Source/BeaconAggregator.pm',
  'INSTALLDIRS' => 'site',
  'EXE_FILES' => [
                   'bin/sasbactrl'
                 ],
  'PREREQ_PM' => {
                   'CGI' => '3.0',
                   'File::Temp' => 0,
                   'SeeAlso::Identifier' => '0.46',
                   'URI::file' => '1.0',
                   'DBI' => '1.57',
                   'Carp' => 0,
                   'LWP' => '5.802',
                   'Test::Command::Simple' => 0,
                   'SeeAlso::Response' => '0.54',
                   'HTTP::Date' => '1.46',
                   'SeeAlso::Identifier::GND' => '0.54',
                   'SeeAlso::Source' => '0.56',
                   'DBD::SQLite' => '1.25',
                   'Test::More' => '0.94',
                   'HTTP::Request' => '1.40',
                   'LWP::UserAgent' => '2.031',
                   'SeeAlso::Server' => '0.57'
                 }
)
;
